src/Bus/TestBus.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Zenstruck\Messenger\Test\Bus;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class TestBus implements MessageBusInterface
+{
+    private MessageBusInterface $decorated;
+    /** @var list<Envelope> */
+    private static array $messages = [];
+
+    // The setting applies to all buses
+    private static bool $enableMessagesCollection = true;
+
+    public function __construct(MessageBusInterface $decorated)
+    {
+        $this->decorated = $decorated;
+    }
+
+    public function dispatched(): BusEnvelope
+    {
+        return new BusEnvelope($this, ...self::$messages);
+    }
+
+    public function dispatch(object $message, array $stamps = []): Envelope
+    {
+        $envelope = $this->decorated->dispatch($message, $stamps);
+
+        if (self::$enableMessagesCollection) {
+            self::$messages[] = $envelope;
```

Comment:
I was thinking about this the other day, and I think we should store the envelopes indexed on the bus' name, the same we we're doing it for the transports.

Otherwise, we couldn't ask for messages which have been dispatched by a specific bus.

```suggestion
            $busName = $envelop->last(BusNameStamp::class)->getBusName() ?? \LogicException('cannot find bus name');
            
            self::$messages[$busName] ??= []
            self::$messages[$busName] = $envelope;
```

and you'll have to impact `TestBus::dispatched()` as well

Reply:
EDIT: actually I think you'll have to store the bus' name inside `TestBus` the same way it is stored in `TestTransport` - so no need to use `BusNameStamp`

Reply:
I'll try to figure out how to add the bus name. Indeed the `AddBusNameStampMiddleware` can be deactivated if I am right.

Reply:
No need to touch the middleware, I was just saying no need to use `BusNameStamp` to get the bus' name

The bus' name is its service id I guess. You can pass the name to the constructor by using `->setArgument('$name', $id)` where you decorate the service